src/diem/testing/miniwallet/client.py

```diff
@@ -118,15 +118,25 @@ def wait_for(self, fn: Callable[[], None], max_tries: int = 120, delay: float =
                 return fn()
             except AssertionError as e:
                 if tries >= max_tries:
-                    events = json.dumps(list(map(self.event_asdict, self.events())), indent=2)
-                    raise TimeoutError("%s, events: \n%s" % (e, events)) from e
+                    raise TimeoutError("%s, events: \n%s" % (e, self.dump_events())) from e
                 time.sleep(delay)
 
+    def log_events(self) -> None:
+        events = self.dump_events()
+        if events:
+            self.client.logger.info("account(%s) events: %s", (self.id, events))
+
+    def dump_events(self) -> str:
+        try:
+            return json.dumps(list(map(self.event_asdict, self.events())), indent=2)
+        except requests.HTTPError:
+            return ""
+
     def event_asdict(self, event: Event) -> Dict[str, Any]:
         ret = asdict(event)
         try:
             ret["data"] = json.loads(event.data)
-            if "kyc_data" in ret["data"]:
+            if ret["data"].get("kyc_data"):
```

Comment:
is `kyc_data` being `None` a valid signal?

Reply:
We made Account#kyc_data optional, so it is possible to be None.
But when it is none, "kyc_data" in ret["data"] may return true, because `{"kyc_data": None}` is a valid output, which cause json decode None failed.
I don't know why previous tests won't cause failure here, probably because this method `event_asdict` only get call when test failed, thus the problem never exposed. I probably need figure out how to write some tests for test failed cases, to ensure util functions like this works as expected.

Reply:
For this event, kyc_data is None, is for the case created_account event, which may contain None as kyc_data for creating account.

Reply:
when this if check returns false, it only means we don't convert data into Dict from a JSON string;
Thus the `kyc_data: None` will be kept in the return value.

Reply:
yeah that was my exact concern :-) thank you